bittensor/btlogging/loggingmachine.py

```diff
@@ -360,45 +361,58 @@ def __trace_on__(self) -> bool:
+    def debug(self, msg="", *args, prefix="", suffix="", **kwargs):
         """Wraps debug message with prefix and suffix."""
-        msg = f"{prefix} - {msg} - {suffix}"
-        self._logger.debug(msg, *args, **kwargs)
+        msg = self._concat_msg(prefix, msg, suffix)
+        self._logger.debug(msg, *args, **kwargs, stacklevel=2)
 
-    def info(self, msg="", prefix="", suffix="", *args, **kwargs):
+    def info(self, msg="", *args, prefix="", suffix="", **kwargs):
         """Wraps info message with prefix and suffix."""
-        msg = f"{prefix} - {msg} - {suffix}"
-        self._logger.info(msg, *args, **kwargs)
+        msg = self._concat_msg(prefix, msg, suffix)
+        self._logger.info(msg, *args, **kwargs, stacklevel=2)
 
-    def success(self, msg="", prefix="", suffix="", *args, **kwargs):
+    def success(self, msg="", *args, prefix="", suffix="", **kwargs):
         """Wraps success message with prefix and suffix."""
-        msg = f"{prefix} - {msg} - {suffix}"
-        self._logger.success(msg, *args, **kwargs)
+        msg = self._concat_msg(prefix, msg, suffix)
+        self._logger.success(msg, *args, **kwargs, stacklevel=2)
 
-    def warning(self, msg="", prefix="", suffix="", *args, **kwargs):
+    def warning(self, msg="", *args, prefix="", suffix="", **kwargs):
         """Wraps warning message with prefix and suffix."""
-        msg = f"{prefix} - {msg} - {suffix}"
-        self._logger.warning(msg, *args, **kwargs)
+        msg = self._concat_msg(prefix, msg, suffix)
+        self._logger.warning(msg, *args, **kwargs, stacklevel=2)
 
-    def error(self, msg="", prefix="", suffix="", *args, **kwargs):
+    def error(self, msg="", *args, prefix="", suffix="", **kwargs):
         """Wraps error message with prefix and suffix."""
-        msg = f"{prefix} - {msg} - {suffix}"
-        self._logger.error(msg, *args, **kwargs)
+        msg = self._concat_msg(prefix, msg, suffix)
+        self._logger.error(msg, *args, **kwargs, stacklevel=2)
 
-    def critical(self, msg="", prefix="", suffix="", *args, **kwargs):
+    def critical(self, msg="", *args, prefix="", suffix="", **kwargs):
         """Wraps critical message with prefix and suffix."""
-        msg = f"{prefix} - {msg} - {suffix}"
-        self._logger.critical(msg, *args, **kwargs)
+        msg = self._concat_msg(prefix, msg, suffix)
+        self._logger.critical(msg, *args, **kwargs, stacklevel=2)
 
-    def exception(self, msg="", prefix="", suffix="", *args, **kwargs):
+    def exception(self, msg="", *args, prefix="", suffix="", **kwargs):
         """Wraps exception message with prefix and suffix."""
-        msg = f"{prefix} - {msg} - {suffix}"
-        self._logger.exception(msg, *args, **kwargs)
+        msg = self._concat_msg(prefix, msg, suffix)
+        stacklevel = 2
+        if (
+            platform.python_implementation() == "CPython"
+            and platform.sys.version_info.major == 3
```

Comment:
I recommend using sys.implementation and sys.version_info for checking the Python implementation and version instead of platform. This approach is more direct, avoids extra imports, and improves performance by leveraging attributes directly available in the sys module, which is standard and efficient for such checks.

Changes:
```python
if (sys.implementation.name == "cpython" and
    sys.version_info.major == 3 and
    sys.version_info.minor < 11):
    # do something
```

Reply:
Done. Syntax is a bit different but as suggested by CI.

Reply:
Perfect. Also pls add unit tests related with new method and updated ones.

Reply:
You mean tests for the logging functionality? They are not part of CI but they all pass. Of the CI tests two tests failed but they seem unrelated: "get current emissions and validate that Alice has gotten tao" and "assert updated_axon.ip == external_ip", could it be that something else is off there? I just pushed a unit test that checks the fixed logging elements.

Reply:
My personal recommendation is to use the following command before sending a PR:
`clear && python3 -m ruff format . && mypy --ignore-missing-imports bittensor/ --python-version=3.9 && mypy --ignore-missing-imports bittensor/ --python-version=3.10 && mypy --ignore-missing-imports bittensor/ --python-version=3.11 && flake8 bittensor/ --count`

I'm surprised that your implementation affected and broke tests that were not related to this. Let's try the following:
- merget staging to your PR (I see `This branch is out-of-date with the base branch` below);
- use locally the command I indicated above;

I expect there will be no broken tests.